src/haddock/clis/cli_analyse.py

```diff
@@ -0,0 +1,631 @@
+        box_plots(ss_file, cluster_ranking)
+
+
+def main(run_dir, modules, top_cluster, **kwargs):
+    """
+    Analyse CLI.
+
+    Parameters
+    ----------
+    run_dir : str or Path
+        Path to the original run directory.
+
+    modules : list of ints
+        List of the integer prefix of the modules to copy.
+
+    top_cluster : int
+        Number of clusters to be considered
+    """
+    log.info(f"Running haddock3-analyse on {run_dir}, modules {modules}, "
+             f"with top_cluster = {top_cluster}")
+
+    # modifying the parameters
+    default_capri = read_from_yaml_config(caprieval_params)
+    capri_dict = default_capri.copy()
+    for param in kwargs:
+        if param not in default_capri:
+            sys.exit(f'* ERROR * Parameter {param!r} is not a valid `caprieval` parameter')  # noqa:E501
+        else:
+            if param.endswith("fname"):  # using full path for files
+                rel_path = Path(kwargs[param])
+                _param = rel_path.resolve()
+                kwargs[param] = _param
+            capri_dict[param] = kwargs[param]
+            log.info(f"setting {param} to {kwargs[param]}")
+
+    os.chdir(run_dir)
+    # Create analysis folder
+    ori_cwd = os.getcwd()
+    outdir = Path(ANA_FOLDER)
+    try:
+        outdir.mkdir(exist_ok=False)
+        log.info(f"Created directory: {str(outdir.resolve())}")
+    except FileExistsError:
+        log.warning(f"Directory {str(outdir.resolve())} already exists.")
+
+    # Reading steps
+    log.info("Reading input run directory")
+    # get the module folders from the run_dir input
+    selected_steps = get_steps(Path("./"), modules)
+    log.info(f"selected steps: {', '.join(selected_steps)}")
+
+    # analysis
+    good_folder_paths, bad_folder_paths = [], []
+    for step in selected_steps:
+        subfolder_name = f"{step}_analysis"
+        target_path = Path(Path("./"), subfolder_name)
+
+        # check if subfolder is already present
+        dest_path = Path(ANA_FOLDER, subfolder_name)
+        if dest_path.exists():
+            if len(os.listdir(dest_path)) != 0:
+                log.warning(f"{dest_path} exists and is not empty. "
+                            "Skipping analysis...")
+                continue
+            else:  # subfolder is empty, remove it.
+                log.info(f"Removing empty folder {dest_path}.")
+                shutil.rmtree(dest_path)
+
+        # run the analysis
+        error = False
+        try:
+            analyse_step(step, Path("./"), capri_dict, target_path, top_cluster)
+        except Exception as e:
+            error = True
+            log.warning(
+                f"""Could not execute the analysis for step {step}.
+                The following error occurred {e}"""
+                )
+        if error:
+            bad_folder_paths.append(target_path)
+        else:
+            good_folder_paths.append(target_path)
+        
+        # going back
+        os.chdir(ori_cwd)
```

Comment:
yes, it is correct! this is related to your other comment above: here the idea is that `analyse_step` is a unique function in which a lot of bad things can potentially happen (errors in the definition of the capri dictionary, errors in the capri execution, errors in the plots). Since the unsuccesful analysis on one step should not kill the overall analysis procedure, I put the `os.chdir(ori_cwd)` command here so that every time we re-execute the commands in the loop we are sure we are in the original cwd directory..is it acceptable?

Reply:
Think its fine!